### ISSUE TITLE
Correctly document ReduceLROnPlateau

### ISSUE THREAD

**Issue by moi90** [PyTorchLightning/pytorch-lightning#1436]
## 📚 Documentation

It seems that [torch.optim.lr_scheduler.ReduceLROnPlateau](https://pytorch.org/docs/stable/optim.html#torch.optim.lr_scheduler.ReduceLROnPlateau) can not be used in `configure_optimizers` because `scheduler.step()` does not receive the loss, see #673, #98 and others.

However, it is used as an example: https://pytorch-lightning.readthedocs.io/en/latest/optimizers.html#learning-rate-scheduling

Furthermore, the required parameter `optimizer` is not passed in the example.

**Comment by moi90** [comment]
The correct usage can be found in the tests: https://github.com/PyTorchLightning/pytorch-lightning/blob/master/tests/base/mixins.py#L698-L705

**Comment by SkafteNicki** [comment]
`ReduceLROnPlateau` can be used, please see here:
https://github.com/PyTorchLightning/pytorch-lightning/blob/7857a73710d8e63eded0e3a970a03b14d84e12f5/pytorch_lightning/trainer/training_loop.py#L775-L785
the `step` function does indeed get conditioned on the value of `monitor_val`.

However, I do agree that the documentation does not make it clear that each learning rate scheduler should receive its corresponding optimizer as input.

**Comment by williamFalcon** [comment]
anyone want to submit a PR for doc updates on this?
